packages/react/src/Token/_RemoveTokenButton.tsx

```diff
@@ -39,71 +44,103 @@ const variants = variant<{height: string; width: string}, TokenSizeKeys>({
-  justify-content: center;
-  align-items: center;
-  user-select: none;
-  appearance: none;
-  text-decoration: none;
-  padding: 0;
-  transform: ${props => `translate(${props.borderOffset}px, -${props.borderOffset}px)`};
-  align-self: baseline;
-  border: 0;
-  border-radius: 999px;
-
-  ${props => {
-    switch (props.size) {
-      case 'large':
-      case 'xlarge':
-        return css`
-          margin-left: ${get('space.2')};
-        `
-      default:
-        return css`
-          margin-left: ${get('space.1')};
-        `
+const StyledTokenButton = toggleStyledComponent(
+  CSS_MODULES_FEATURE_FLAG,
+  'span',
+  styled.span<TokenButtonProps>`
+    background-color: transparent;
+    font-family: inherit;
+    color: currentColor;
+    cursor: pointer;
+    display: inline-flex;
+    justify-content: center;
+    align-items: center;
+    user-select: none;
+    appearance: none;
+    text-decoration: none;
+    padding: 0;
+    transform: ${props => `translate(${props.borderOffset}px, -${props.borderOffset}px)`};
+    align-self: baseline;
+    border: 0;
+    border-radius: 999px;
+
+    ${props => {
+      switch (props.size) {
+        case 'large':
+        case 'xlarge':
+          return css`
+            margin-left: ${get('space.2')};
+          `
+        default:
+          return css`
+            margin-left: ${get('space.1')};
+          `
+      }
+    }}
+
+    &:hover,
+    &:focus {
+      // TODO: choose a better functional color variable for this
+      background-color: ${get('colors.neutral.muted')};
     }
-  }}
 
-  &:hover,
-  &:focus {
-    // TODO: choose a better functional color variable for this
-    background-color: ${get('colors.neutral.muted')};
-  }
+    &:active {
+      // TODO: choose a better functional color variable for this
+      background-color: ${get('colors.neutral.subtle')};
+    }
 
-  &:active {
-    // TODO: choose a better functional color variable for this
-    background-color: ${get('colors.neutral.subtle')};
-  }
+    ${variants}
+    ${sx}
+  `,
+)
 
-  ${variants}
-  ${sx}
-`
+type RemoveTokenButtonProps = TokenButtonProps & Omit<React.HTMLProps<HTMLSpanElement | HTMLButtonElement>, 'size'>
 
-const RemoveTokenButton: React.FC<React.PropsWithChildren<ComponentProps<typeof StyledTokenButton>>> = ({
+const RemoveTokenButton = ({
   'aria-label': ariaLabel,
   isParentInteractive,
   size = defaultTokenSize,
+  className,
   ...rest
-}) => {
+}: React.PropsWithChildren<RemoveTokenButtonProps>) => {
   delete rest.children
 
+  const enabled = useFeatureFlag(CSS_MODULES_FEATURE_FLAG)
+
   return (
     <StyledTokenButton
       as={isParentInteractive ? 'span' : 'button'}
       tabIndex={isParentInteractive ? -1 : undefined}
       aria-label={!isParentInteractive ? 'Remove token' : ariaLabel}
       size={size}
+      data-size={size}
+      className={clsx(
+        enabled && classes.TokenButton,
+        enabled && addLargerMarginClass(size) && classes.Bigger,
```

Comment:
```suggestion
```

Reply:
ugh, 🤦 thank you for catching that

Reply:
[Removed](https://github.com/primer/react/pull/5251/commits/06280034904ad7b133890b4324711798c1e5632d)